assets/app/scripts/controllers/newfromtemplate.js

```diff
@@ -39,6 +39,11 @@ angular.module('openshiftConsole')
       }
     ];
 
+    // In case user wants just to instantiate the template, without creating it in the 'Add to Project'
+    // process, get the template from the 'TemplateServide'
+    $scope.template = TemplateService.getTemplate();
+    TemplateService.clearTemplate();
```

Comment:
I suggest using $routeParameters to tell whether to process a template from TemplateService or use one that is in a namespace.